### PR TITLE
Test: Unit test for update training job

### PR DESCRIPTION
**Pull request by ananth102**
Description of changes:

Merge after #179 

Unit tests for update training job.

File changes:
`sagemaker-controller/pkg/resource/training_job/testdata/v1alpha1/update/*` - Test files
`sagemaker-controller/pkg/resource/training_job/testdata/sdkapi/update` - Mock api responses
`sagemaker-controller/pkg/resource/training_job/testdata/test_suite.yaml` - Test cases
`sagemaker-controller/pkg/resource/training_job/manager_test_suite_test.go` - Change to support update API

### CODE REVIEW

File: pkg/resource/training_job/testdata/sdkapi/update/update_sucess.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (ryansteakley) [inline pkg/resource/training_job/testdata/sdkapi/update/update_sucess.json:1]: nit: success is spelled incorrectly

File: pkg/resource/training_job/testdata/test_suite.yaml [hunk 2]
```diff
@@ -210,4 +249,196 @@ tests:
...
+            - operation: UpdateTrainingJobWithContext
+              output_fixture: "sdkapi/update/update_sucess.json"
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/update/describe/profiler_removal.json"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/removeProfilerRule.yaml"
+          error: training job is updating
+      - name: "Update=RemoveProfilerConfig"
+        desciption: "This test checks if the Controller can remove the profiler properly"
```
Comment (ryansteakley) [inline pkg/resource/training_job/testdata/test_suite.yaml:285]: nit: looks like description is spelled incorrectly here for all description fields, if oyu have time do you mind making the descriptions have more detail for the specific test. We can infer the reason from the test-name but a lot ofthese have the same description

Comment (ananth102) [reply pkg/resource/training_job/testdata/test_suite.yaml:285]: Yea I will update them for completeness

File: pkg/resource/training_job/testdata/test_suite.yaml [hunk 3]
```diff
@@ -210,4 +249,196 @@ tests:
...
+            - operation: UpdateTrainingJobWithContext
+              output_fixture: "sdkapi/update/update_sucess.json"
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/update/describe/profiler_config_update.json"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/updateProfilerConfig.yaml"
+          error: training job is updating
+      - name: "Update=ProfilerStarting"
+        description: "Tests if profiler returns a requeue error when TJ is updated in Starting."
```
Comment (ryansteakley) [inline pkg/resource/training_job/testdata/test_suite.yaml:327]: by TJ you are referring to training-job or some field?

Comment (ananth102) [reply pkg/resource/training_job/testdata/test_suite.yaml]: Training Job

**Review by ryansteakley — APPROVED** [review]

**Pull request merged**